dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdNdkSymbolFileUploadTask.kt

```diff
@@ -0,0 +1,191 @@
+                        file,
+                        encoding = ENCODING,
+                        TYPE_NDK_SYMBOL_FILE,
+                        file.name,
+                        mapOf(
+                            "arch" to arch
+                        )
+                    )
+                )
+            }
+
+        return files
+    }
+
+    private fun findSoFiles(searchDirectory: File): Collection<File> {
+        return if (searchDirectory.exists() && searchDirectory.isDirectory) {
+            searchDirectory.walkTopDown()
+                .filter { it.extension == "so" }
+                .toSet()
+        } else {
+            return emptySet()
+        }
+    }
+
+    companion object {
+        internal const val TASK_NAME = "ddUploadNdkSymbolFiles"
+        internal const val KEY_NDK_SYMBOL_FILE = "ndk_symbol_file"
+        internal const val TYPE_NDK_SYMBOL_FILE = "ndk_symbol_file"
+        internal const val ENCODING = "application/octet-stream"
+        internal val SUPPORTED_ARCHS = setOf("armeabi-v7a", "arm64-v8a", "x86", "x86_64")
+        internal const val MAX_DATADOG_CI_FILE_LOOKUP_LEVELS = 4
+
+        internal val LOGGER = Logging.getLogger("DdSymbolFileUploadTask")
+
+        @Suppress("StringLiteralDuplication")
+        private fun resolveDatadogRepositoryFile(target: Project): File {
+            val outputsDir = File(target.buildDir, "outputs")
+            val reportsDir = File(outputsDir, "reports")
+            val datadogDir = File(reportsDir, "datadog")
+            return File(datadogDir, "repository.json")
+        }
+
+        internal fun findDatadogCiFile(projectDir: File): File? {
+            var currentDir: File? = projectDir
+            var levelsUp = 0
+            while (currentDir != null && levelsUp < MAX_DATADOG_CI_FILE_LOOKUP_LEVELS) {
+                val datadogCiFile = File(currentDir, "datadog-ci.json")
+                if (datadogCiFile.exists()) {
+                    return datadogCiFile
+                }
+                currentDir = currentDir.parentFile
+                levelsUp++
+            }
+            return null
+        }
+
+        @Suppress("MagicNumber", "ReturnCount")
+        private fun isAgp7OrAbove(): Boolean {
+            val version = Version.ANDROID_GRADLE_PLUGIN_VERSION
+            val groups = version.split(".")
+            if (groups.size < 3) return false
+            val major = groups[0].toIntOrNull()
+            val minor = groups[1].toIntOrNull()
+            val patch = groups[2].substringBefore("-").toIntOrNull()
+            if (major == null || minor == null || patch == null) return false
+            return major >= 7 && minor >= 0 && patch >= 0
+        }
+
+        private fun fixNativeOutputPath(taskFolder: File): File {
+            return taskFolder.parentFile.parentFile.takeIf { it.parentFile.name == "cxx" }
+                ?: taskFolder.parentFile.parentFile.parentFile.takeIf { it.parentFile.name == "cxx" }
```

Comment:
How can we be sure that `parentFile` always non-null? I think this API is coming from Java, so Kotlin cannot validate if it is null or non-null, but essentially, to prevent NPE we need either do the check or `?.` calls after `parentFile`

Reply:
I'm removing this. It doesn't seem necessary. I'm sure we'll learn why it's there in the future.